include/zephyr/shell/shell.h

```diff
@@ -91,15 +91,12 @@ typedef void (*shell_dynamic_get)(size_t idx,
 /**
  * @brief Shell command descriptor.
  */
-struct shell_cmd_entry {
-	bool is_dynamic;
-	union union_cmd_entry {
-		/*!< Pointer to function returning dynamic commands.*/
-		shell_dynamic_get dynamic_get;
-
-		/*!< Pointer to array of static commands. */
-		const struct shell_static_entry *entry;
-	} u;
+union shell_cmd_entry {
```

Comment:
We could keep the `struct shell_cmd_entry` having just the union in it for possible future extensions?

Reply:
I meant without having to touch each and every user of the union/struct at that time.